### PR TITLE
HowardHinnant-Date 2.4.1 (new formula)

### PR DESCRIPTION
**Pull request by cdalvaro**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

This formula is for @HowardHinnant date library
-----

### CODE REVIEW

File: Formula/date.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
+class Date < Formula
+  desc "C++11/14/17 library for Date and Time operations based on <chrono> header"
+  homepage "https://github.com/HowardHinnant/date"
+  url "https://github.com/HowardHinnant/date/archive/v2.4.1.tar.gz"
+  sha256 "98907d243397483bd7ad889bf6c66746db0d7d2a39cc9aacc041834c40b65b98"
+
+  option "with-system-tz-db", "Use the operating system's timezone database"
+  option "without-shared", "Build a static version of library"
```
Comment (fxcoudert) [inline Formula/date.rb]: No options for new formulas. Please choose the “most common” feature set that users will want, and restrict to that.

File: Formula/date.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    end
+
+    if build.without? "shared"
+      arcustom_cmake_argsgs << "-DBUILD_SHARED_LIBS=OFF"
+    else
+      custom_cmake_args << "-DBUILD_SHARED_LIBS=ON"
+    end
+
+    system "cmake", ".", *std_cmake_args, *custom_cmake_args
+    system "make", "-j#{ENV.make_jobs}"
```
Comment (fxcoudert) [inline Formula/date.rb]: Let Homebrew handle the `-j` stuff.

File: Formula/date.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    system "make", "install"
```
Comment (fxcoudert) [inline Formula/date.rb]: `make install` is probably working without doing a separate `make`

File: Formula/date.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  end
+
+  test do
+    (testpath/"test.cpp").write <<~EOS
+      #include "date/date.h"
+      #include <type_traits>
+      static_assert(std::is_same<decltype(date::last), const date::last_spec>{}, "");
```
Comment (fxcoudert) [inline Formula/date.rb]: Is this a header-only library? If not, we need to check that it links.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

**Review by ilovezfs — DISMISSED** [review]
this'll need a far less generic name

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/cppdate.rb [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  desc "C++11/14/17 library for Date and Time operations based on <chrono>"
+  homepage "https://github.com/HowardHinnant/date"
+  url "https://github.com/HowardHinnant/date/archive/v2.4.1.tar.gz"
+  sha256 "98907d243397483bd7ad889bf6c66746db0d7d2a39cc9aacc041834c40b65b98"
+
+  depends_on "cmake" => :build
+  needs :cxx11
+
+  def install
+    custom_cmake_args = ["-DENABLE_DATE_TESTING=OFF",
```
[the remark this answers is not in the document]
Comment (cdalvaro) [reply Formula/cppdate.rb]: Done!

File: Formula/cppdate.rb [hunk 6]
```diff
@@ -0,0 +1,32 @@
+class Cppdate < Formula
+  desc "C++11/14/17 library for Date and Time operations based on <chrono>"
```
[the remark this answers is not in the document]
Comment (cdalvaro) [reply Formula/cppdate.rb]: Done!

File: Formula/cppdate.rb [hunk 7]
```diff
@@ -0,0 +1,32 @@
+class Cppdate < Formula
```
[the remark this answers is not in the document]
Comment (cdalvaro) [reply Formula/cppdate.rb]: Done!

**Review by cdalvaro — COMMENTED** [review]
All requested changes have been performed

**Comment by cdalvaro** [comment]
ping @fxcoudert @ilovezfs

**Review by cdalvaro — COMMENTED** [review]
I did all requested changes:

* [x] No options for new formulas
* [x] Let Homebrew handle the `-j` stuff
* [x] Combine `make all` and `make install`
* [x] Modified the test to link against the installed library
* [x] Changed formula name to `HowardHinnant-Date`

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
Thanks @cdalvaro for the pull request!

**Pull request closed without merging**